crates/cheatcodes/src/evm.rs

```diff
@@ -89,7 +90,13 @@ impl Cheatcode for loadCall {
         let Self { target, slot } = *self;
         ensure_not_precompile!(&target, ccx);
         ccx.ecx.load_account(target)?;
-        let val = ccx.ecx.sload(target, slot.into())?;
+        let mut val = ccx.ecx.sload(target, slot.into())?;
+        // Generate random value if target should have arbitrary storage and storage slot untouched.
+        if ccx.state.arbitrary_storage.contains(&target) && val.is_cold && val.data.is_zero() {
```

Comment:
what's considered "untouched" here? eg in a single test calling the same contract two times from test scope would result in slot being cold on first call, and warm on second due to how our EVM works. is this expected?

Reply:
yep, that's the expected behavior, subsequent loads of same slot should return same value

https://github.com/foundry-rs/foundry/blob/10748dc2d5235a5cef4e8077a3b7bf475e1ef17b/testdata/default/cheats/ArbitraryStorage.t.sol#L108

unless explicitly rewritten

https://github.com/foundry-rs/foundry/blob/10748dc2d5235a5cef4e8077a3b7bf475e1ef17b/testdata/default/cheats/ArbitraryStorage.t.sol#L51